stuffer/s2n_stuffer.c

```diff
@@ -437,7 +441,13 @@ int s2n_stuffer_shift(struct s2n_stuffer *stuffer)
     POSIX_ENSURE_REF(stuffer);
     struct s2n_stuffer copy = *stuffer;
     POSIX_GUARD(s2n_stuffer_rewrite(&copy));
-    uint8_t *data = stuffer->blob.data + stuffer->read_cursor;
+
+    /* Adding 0 to a NULL value is undefined behavior */
+    uint8_t *data = stuffer->blob.data;
+    if (stuffer->read_cursor != 0) {
```

Comment:
Nit: I think the comment belongs on the if, not on the declaration (same in early data), but it doesn't super matter.
```suggestion
    uint8_t *data = stuffer->blob.data;
    /* Adding 0 to a NULL value is undefined behavior */
    if (stuffer->read_cursor != 0) {
```